### ISSUE TITLE
PSR12: should PSR2.Classes.PropertyDeclaration.Underscore be reported as an error?

### ISSUE THREAD

**Issue by rodrigoprimo** [PHPCSStandards/PHP_CodeSniffer#1426]
The PSR12 ruleset includes a severity override that promotes `PSR2.Methods.MethodDeclaration.Underscore` from warning to error and customises the message text from "should not" to "must not", matching the `MUST NOT` wording in [PSR-12 section 4.4](https://www.php-fig.org/psr/psr-12/#44-methods-and-functions).

The equivalent rule for properties in [PSR-12 section 4.3](https://www.php-fig.org/psr/psr-12/#43-properties-and-constants) uses identical `MUST NOT` wording, but no matching override exists for `PSR2.Classes.PropertyDeclaration.Underscore`. As a result, a property with a leading underscore is reported as a warning, while a method with a leading underscore is reported as an error.

Both were introduced in b7fed6b6, but only the method override was applied.

Given the following code sample:

```php
<?php

namespace MyNamespace;

class MyClass
{
    public $_foo;
    protected function _bar()
    {
    }
}
```

Running `phpcs --standard=PSR12` against this file produces:

```
7 | WARNING | Property name "$_foo" should not be prefixed with an underscore to indicate visibility
8 | ERROR   | Method name "_bar" must not be prefixed with an underscore to indicate visibility
```

I wonder if there's a reason I'm missing to treat the property rule as a warning or if the PSR12 ruleset should be updated? If the latter, I can open a PR suggesting this change.

**Comment by jrfnl** [comment]
> I wonder if there's a reason I'm missing to treat the property rule as a warning or if the PSR12 ruleset should be updated? If the latter, I can open a PR suggesting this change.

I can't find a reason, nor can I find anything of relevance in the file histories (ruleset/sniff) to explain why this happened.

I agree that for PSR12, the `PSR2.Classes.PropertyDeclaration.Underscore` message should be promoted to an error and the message should probably also be tweaked a little to be in line with PSR12.

Thanks for finding this oversight @rodrigoprimo !

Marking this as a "good first issue".

**Comment by syedusmanali** [comment]
Picking this up, PR coming shortly.
